apps/backoffice/src/app/api/auth/[...nextauth]/auth.ts

```diff
@@ -65,21 +68,90 @@ const retrieveApimUser = (config: Configuration) => (
   pipe(
     ApimUtils.getApimClient(config, config.AZURE_SUBSCRIPTION_ID),
     apimClient =>
-      ApimUtils.getApimService(
-        apimClient,
-        config.AZURE_APIM_RESOURCE_GROUP,
-        config.AZURE_APIM
-      ),
-    apimService =>
-      apimService.getUserByEmail(
-        `org.${identityTokenPayload.organization.id}@selfcare.io.pagopa.it` as EmailString,
-        true
+      pipe(
+        ApimUtils.getApimService(
+          apimClient,
+          config.AZURE_APIM_RESOURCE_GROUP,
+          config.AZURE_APIM
+        ),
+        apimService =>
+          apimService.getUserByEmail(
+            formatApimAccountEmailForSelfcareOrganization(
+              identityTokenPayload.organization
+            ),
+            true
+          ),
+        TE.mapLeft(
+          err =>
+            new Error(
+              `Failed to fetch user by its email, code: ${err.statusCode}`
+            )
+        ),
+        TE.chain(
+          flow(
+            O.fold(
+              () =>
+                pipe(
+                  TE.tryCatch(
+                    () =>
+                      apimClient.user.createOrUpdate(
```

Comment:
Non Blocking, this method can be added to external-service/apim-client in order to have all interaction with Apim in a single place

Reply:
@giamma1295 I know, but I would like to introduce no more changes than are strictly necessary (related to this task). I will handle this refactor in a dedicated PR 😉